### PR TITLE
FlxCamera: fix object visibility for zoom < 1, closes #292

### LINKED ISSUE #292: Fix object visibility tests for cases when camera have zoom less than 1.0
You can see this issue in FlxCamera Demo ( http://haxeflixel.com/demos/flxcamera )
##

### PR DESCRIPTION
**Pull request by Beeblerox**
Support for camera zoom values less than 1 for blit and tile render modes

### CODE REVIEW

**Comment by Beeblerox** [comment]
@Gama11 could you review this pull request please?

**Comment by Gama11** [comment]
On Flash, the FlxCamera demo has an issue it didn't have before. If you zoom out all the way to 0.5x and then back to 1x, you get this:



Note the black borders at the edges.

**Comment by Beeblerox** [comment]
@Gama11 i'll try to see what's wrong with this

**Comment by Beeblerox** [comment]
@Gama11 i think i've fixed this issue with FlxCamera demo. At least it works as expected on my machine. Could you check it also?

**Comment by starry-abyss** [comment]
It looks like @Gama11 's issue was fixed. Also my demo is working too.
But I see that tilemap tearing on Flash in the FlxCamera demo became more frequent at zooms < 1, comparing to dev branch

**Comment by Beeblerox** [comment]
@starry-abyss FlxCamera demo doesn't use tilemaps. Actually you see gaps between FlxSprites which forms background

**Comment by Beeblerox** [comment]
i just can't fix tearing problem on flash target for zoom values less than 1.0. and doubt that this is possible.
Gonna leave it as it is now

**Comment by gamedevsam** [comment]
Is this change compatible with new opengl renderer?

**Comment by Beeblerox** [comment]
@gamedevsam i've merged this changes on my working branch recently

**Comment by Gama11** [comment]
So should this PR be closed?

**Comment by Gama11** [comment]
@Beeblerox This PR adds some TODO comments, are these still relevant?

**Comment by starry-abyss** [comment]
What is the reason to not merge it before the OpenFl 4 PR?

File: flixel/FlxCamera.hx [hunk 1]
```diff
@@ -205,6 +201,44 @@ class FlxCamera extends FlxBasic
...
+	 * Difference between native size of camera and zoomed size, divided in half
+	 * Needed to do occlusion of objects when zoom != initialZoom
+	 */
+	public var viewOffsetX(default, null):Float = 0;
+	public var viewOffsetY(default, null):Float = 0;
+	
+	/**
+	 * Floored down values of viewOffsetX and viewOffsetY.
+	 */
+	private var viewOffsetXFloored:Int = 0;
```
Comment (Beeblerox) [inline flixel/FlxCamera.hx]: @Gama11 ok, i'll remove these variables.

File: flixel/FlxCamera.hx [hunk 2]
```diff
@@ -713,6 +781,85 @@ class FlxCamera extends FlxBasic
 	}
 	
 	/**
+	 * Helper method preparing debug rectangle for rendering in blit render mode
+	 * @param	rect	rectangle to prepare for rendering
+	 * @return	trasformed rectangle with respect to camera's zoom factor
+	 */
+	@:noCompletion
```
Comment (Gama11) [inline flixel/FlxCamera.hx]: I'm not a big fan of `@:noCompletion`, things still show up in API docs with that (see [`startQuadBatch()`](http://api.haxeflixel.com/flixel/FlxCamera.html?#startQuadBatch) for instance). There's `@:dox(hide)`, but I feel like it's cleaner to just make them `private` to begin with (and use `@:access` to access them)?

**Pull request merged**